src/sitepotentials.jl

```diff
@@ -184,11 +184,13 @@ function AtomsCalculators.energy_forces!(f::AbstractVector, at, V::SitePotential
       Js, Rs, Zs, z0 = get_neighbours(at, V, nlist, i) 
       v, dv = eval_grad_site(V, Rs, Zs, z0)
       E += v 
-      f[Js] -= dv * force_unit(V)
-      f[i]  += sum(dv) * force_unit(V)
+      for α = 1:length(Js)
+         f[Js[α]] -= dv[α] * force_unit(V)
+         f[i]     += dv[α] * force_unit(V)
```

Comment:
It is better to not combine these two into one loop.

`f[Js[α]]`  is a random access on an array and is thus slow. This is faster in many cases by just having

```julia
for (α, j) in enumerate(Js)
         f[j] -= dv[α] * force_unit(V)
end
f[i] += sum(dv) * force_unit(V)
```

Reply:
I think the performance difference will not be measurable in the current generation of codes which have an enourmous overhead all over the place.

Reply:
For readability I don't want to move the `f[i]` outside the loop. If you can find me an actual use-case where this matters then we can revisit this point. I suspect that this won't happen since the current codes have far too much overhead.